R/data_type.R

```diff
@@ -124,7 +124,7 @@ identify_data_type <- function(df,
     df[matching_idx, "type"]  <- ifelse(
       is.na(df[matching_idx, "type"]), 
       type, 
-      df[matching_idx, "type"]
+      unlist(df[matching_idx, "type"])
```

Comment:
You don't need to use `unlist` in data.table if you want to represent single column as a vector. Just simply get rid of quotation marks: `df[matching_idx, type]`